### PR TITLE
Temporarily display the plans in the inspector.

### PR DESCRIPTION
**Pull request by huwshimi**
DEMO WARE!

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5476/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5477/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5478/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5479/
Test PASSed.

**Comment by anthonydillon** [comment]
Code looks good :+1: even if it's only demo ware

**Comment by anthonydillon** [comment]
Will QA once I get to the office. If someone doesn't beat me to it.

File: jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js]: Please add a top level docstring describing the goal of this module.

File: jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          this.props.service.get('charm'), this._getPlansCallback);
+      })  ;
+    },
+
+    /**
+      Callback for when plans for an entity have been successfully fetched.
+
+      @method _getPlansCallback
+      @param {String} error An error message, or null if there's no error.
+      @param {Array} models A list of the plans found.
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js]: s/models/plans/
Also please specify available fields on a plan object.

File: jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    },
+
+    /**
+      Get the list of plans available for the service.
+
+      @method _getPlans
+    */
+    _getPlans: function() {
+      this.setState({plansLoading: true}, () => {
+        this.plansXHR = this.props.listPlansForCharm(
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js]: In theory the list of plans could be already present as an attribute of the charm. retrieved when the charm detail was loaded. So we could save a roundtrip in that case. But not for this branch, maybe add a TODO?

File: jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+          }
+        }
+      });
+    },
+
+    /**
+      Generate a list of plans.
+
+      @method _generatePlans
+      @return {Function} The React elements for the UI.
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector-plan/plans/plans.js]: The React elements is a function?

File: jujugui/static/gui/src/app/components/inspector/inspector.js [hunk 5]
```diff
@@ -415,6 +418,27 @@ YUI.add('inspector-component', function() {
                   activeComponent: undefined
                 }}}};
           break;
+        case 'plans':
+          // DEMO-WARE DEMO-WARE DEMO-WARE DEMO-WARE DEMO-WARE DEMO-WARE!
+          // If you see this code and it's after July 2016 you can remove this
+          // and the component. Seriously, just do it.
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/inspector/inspector.js]: "Yesterday you said tomorrow, so just do it!"

**Comment by frankban** [comment]
Hey Huw, the code is OK with same small comment that can be addressed later.
QA is OK even if this is not exactly what I expected (explained in IRC)

**Comment by huwshimi** [comment]
Thanks everyone :shipit:!

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5482/
Test PASSed.

**Pull request merged**